src/osd/ScrubStore.cc

```diff
@@ -0,0 +1,159 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+
+#include "ScrubStore.h"
+#include "osd_types.h"
+#include "common/scrub_types.h"
+#include "include/rados/rados_types.hpp"
+
+namespace {
+hobject_t make_scrub_object(const spg_t& pgid)
+{
+  ostringstream ss;
+  ss << "scrub_" << pgid;
+  return pgid.make_temp_object(ss.str());
+}
+
+string first_object_key(int64_t pool)
+{
+  return "SCRUB_OBJ_" + std::to_string(pool) + "-";
+}
+
+// the object_key should be unique across pools
+string to_object_key(int64_t pool, const librados::object_id_t& oid)
+{
+  return ("SCRUB_OBJ_" +
+	  std::to_string(pool) + "." +
+	  oid.name + oid.nspace + std::to_string(oid.snap));
```

Comment:
This is problematic because you could have two objects in the same pg:
1) name: foo namespace: bar
2) name: foob namespace: ar
We need a seperator.  I think there are other places in the code where object names are used to generate keys, how about using hobject_t::to_str()?